### PR TITLE
Implement variable dependencies in the questionnaire UI

### PR DESCRIPTION
**Pull request by nickgraz**
- Updated types from BE
- Add dependency check for variable visibility

Currently supported dependencies (we may need to add more):
- Number type variable - **=, >=, >, <, <=, !=**
- Text type variable - **=, !=**
- Select type variable - **= (in), != (not in)**

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#2835** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2835?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/scenes/DeliverablesRouter/QuestionsDeliverableView.tsx [hunk 1]
```diff
@@ -187,9 +188,11 @@ const QuestionsDeliverableView = (props: Props): JSX.Element | null => {
                 paddingTop: theme.spacing(3),
               }}
             >
-              {variablesWithValues.map((variableWithValues: VariableWithValues, index: number) => (
-                <QuestionBox key={index} projectId={projectId} variable={variableWithValues} />
-              ))}
+              {variablesWithValues.map((variableWithValues: VariableWithValues, index: number) =>
+                variableDependencyMet(variableWithValues, variablesWithValues) ? (
```
Comment (constanzauanini) [inline src/scenes/DeliverablesRouter/QuestionsDeliverableView.tsx:192]: Should we validate this on the external side too?

**Review by constanzauanini — APPROVED** [review]
Lgtm. Just one question

File: src/scenes/DeliverablesRouter/QuestionsDeliverableView.tsx (hunk 1 above)
Comment (joshbuchea) [reply src/scenes/DeliverablesRouter/QuestionsDeliverableView.tsx:192]: So this view is for the external (participant) side, but still a good question. I'm not sure if there's a requirement to conditionally render the questions on the internal (accelerator) side.

Comment (nickgraz) [reply src/scenes/DeliverablesRouter/QuestionsDeliverableView.tsx:192]: I'll make another PR for the internal side

Comment (joshbuchea) [reply src/scenes/DeliverablesRouter/QuestionsDeliverableView.tsx:192]: Do we need to conditionally render questions still on the internal side? I could see the case for internal users wanting to see all of the questions, regardless of the current answer/value status.

Comment (nickgraz) [reply src/scenes/DeliverablesRouter/QuestionsDeliverableView.tsx:192]: Good question, I'll check to see if there's a clear answer in the PRD

**Pull request merged**